pkg/addons/adminconsole/adminconsole.go

```diff
@@ -86,81 +104,156 @@ func (a *AdminConsole) addLicenseToHelmValues() error {
 	return nil
 }
 
-func (a *AdminConsole) Apply(ctx context.Context) error {
-	version, err := a.Latest()
-	if err != nil {
-		return fmt.Errorf("unable to get latest Admin Console version: %w", err)
+// GetPasswordFromConfig returns the adminconsole password from the cluster config.
+func (a *AdminConsole) GetPasswordFromConfig() (string, error) {
+
+	chartList := a.config.Spec.Extensions.Helm.Charts
```

Comment:
Can't Spec be nil an then this will segfault ? There may be other nullable properties inside Spec.